packages/@sanity/types/src/schema/types.ts

```diff
@@ -172,7 +177,12 @@ export type SchemaValidationValue =
   | ((rule: Rule) => SchemaValidationValue)
 
 /** @public */
-export interface BaseSchemaType {
+export interface DeprecatedSchemaType {
```

Comment:
Would it be better to make this type generic? I'm thinking something like:

```ts
// update the base schema type to include an optional `deprecated` prop
interface BaseSchemaType {
  // ... existing interface
  deprecated?: DeprecatedProperty
}

// update the DeprecatedSchemaType to be generic but include a default
type DeprecatedSchemaType<TSchemaType extends BaseSchemaType = BaseSchemaType> =
  TSchemaType & Required<Pick<TSchemaType, 'deprecated'>>

// update the narrowing function to be generic as well
function isDeprecatedSchemaType<TSchemaType extends BaseSchemaType>(type: TSchemaType): type is DeprecatedSchemaType<TSchemaType> {
  // ...
}
```

This way if you ever use the `DeprecatedSchemaType` type alone, it would include all the properties of BaseSchemaType and the resulting type after a narrowing would be `DeprecatedSchemaType<ObjectSchemaType>` instead of `DeprecatedSchemaType & ObjectSchemaType`

Reply:
Thanks Rico, good point! I've adopted your feedback in [this commit](https://github.com/sanity-io/sanity/pull/5489/commits/cfb65c0b37208f662e3c1f7ad484708167dcd01b). I've added a `isDeprecationConfiguration` type guard, because in one instance we were checking whether an object that was not necessarily a schema contained this configuration.